### DISCUSSION TITLE
FetchError: invalid json response body at <URL> reason: Unexpected token T in JSON at position 0

### DISCUSSION THREAD

**Discussion by 0jg** [vercel/vercel#4732] [Help]
Hey!

I'm having a hard time deploying to Preview with Vercel. Local development builds fine, and all API requests work as expected. However, when deploying to Vercel, the API hits an error and obviously doesn't return valid JSON (whereas in development it does).

```
> Build error occurred
FetchError: invalid json response body at https://dev.opn.ac/api/journal reason: Unexpected token T in JSON at position 0
    at /vercel/3be3e940/node_modules/next/dist/compiled/node-fetch/index.js:1:133590
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async getStaticPaths (/vercel/3be3e940/.next/serverless/pages/j/[slug]/[id].js:871:16)
    at async buildStaticPaths (/vercel/3be3e940/node_modules/next/dist/build/utils.js:18:80)
    at async Object.isPageStatic (/vercel/3be3e940/node_modules/next/dist/build/utils.js:25:542) {
  type: 'invalid-json'
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Error: Command "yarn run build" exited with 1
```

Is this a time out error? Sadly there's no further information and I can't find any mechanism to log directly to the build logs (the API works fine on development)

Cheers,
Jack

**Comment by styfle** [comment]
I'm seeing a 404 from https://dev.opn.ac/api/journal

<img src="https://user-images.githubusercontent.com/229881/85958727-9a502300-b965-11ea-8a8a-04cf41a4303e.png" 
 width="500" />

> Local development builds fine, and all API requests work as expected

Is 404 the intended response?

**Reply by 0jg** [reply]
Hey! No—this should resolve fine. I believe that because the actual build fails there's no actual API endpoint there. Currently, I have the absolute URL for a `fetch()` request pointing to `http://dev.opn.ac/api/journal`, where `http://dev.opn.ac` is set depending on the `NODE_ENV` environment variable (localhost if development). Should this API endpoint be made available somehow prior to deployment? Building the API endpoint itself is part of this entire build process.

**Reply by styfle** [reply]
>  Should this API endpoint be made available somehow prior to deployment?

It sounds like it is necessary during build if you are calling fetch() during build time

**Reply by 0jg** [reply]
Hmmm—would you suggest a deployment of the frontend to Vercel, and then a deployment of the API separately? Seems a bit of an odd implementation... surely API endpoints should also be built alongside?

**Reply by styfle** [reply]
Perhaps you should create a utility function that is shared between frontend and backend, no need to create a HTTP request.
